probe/pkg/runtime/runtime.go

```diff
@@ -0,0 +1,64 @@
+package runtime
+
+import (
+	"context"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"github.com/stackrox/acs-fleet-manager/probe/config"
+	"github.com/stackrox/acs-fleet-manager/probe/pkg/probe"
+	"github.com/stackrox/rox/pkg/utils"
+)
+
+// Runtime performs a probe run against fleet manager.
+type Runtime struct {
+	Config *config.Config
+}
+
+// New creates a new runtime.
+func New() (*Runtime, error) {
+	config, err := config.GetConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to load configuration")
+	}
+
+	return &Runtime{
+		Config: config,
+	}, nil
+}
+
+// RunLoop a continuous loop of probe runs.
+func (r *Runtime) RunLoop(ctx context.Context) error {
+	ticker := time.NewTicker(r.Config.RuntimeRunWaitPeriod)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return errors.Wrap(ctx.Err(), "probe context invalid")
+		case <-ticker.C:
+			if err := r.RunSingle(ctx); err != nil {
+				glog.Warning(err)
+			}
+		}
+	}
+}
+
+// RunSingle executes a single probe run.
+func (r *Runtime) RunSingle(ctx context.Context) error {
+	ctxTimeout, cancel := context.WithTimeout(ctx, r.Config.RuntimeRunTimeout)
+	defer cancel()
+	defer utils.Should(r.CleanUp())
```

Comment:
`utils.Should` should only be used when you think that an `error` can only occur because of a programming error. I think cleaning up can fail for a variety of reasons (ACSMS down?) that are not programming errors.
```suggestion
	defer r.CleanUp()
```